specs/impact-graph.md

```diff
@@ -0,0 +1,223 @@
+# Impact Graph: A Software Impact Model
+
+If you can’t measure, you can’t improve. Software has many negative environmental **impacts** which we need to optimize, carbon, water, energy to name just a few.
+
+Unfortunately, measuring software impact metrics like carbon, water, energy is complex and very nuanced. 
+
+Modern applications are composed of many smaller pieces of software (components) running on many different environments, for example, private cloud, public cloud, bare-metal, virtualized, containerized, mobile, laptops, desktops, embedded, IoT. Many components that make up a typical software application are not run on resources you own or control, which makes including the impact of managed services in your measurement especially hard. 
+
+- How we model the environmental impacts of a complex application is through the creation of an **Impact Graph** which represents and models the thing we want to measure. 
+- Each **Impact Graph Node** in the graph represents one atomic thing we want to measure, it could be a software component, or part of the user joruney. 
+- To measure the environmental impact of one of the nodes in an **Impact Graph** we use a **Impant Mesurement Model**. 
+- There are many **Impact Measurement Models** out there and they all have different interfaces, some are APIs some are libraries, some are just CSV files. 
+- Since every **Impact Measurement Model** is bespoke to make them all interoperable we write **Impact Measurement Plugins** which adhere to a common **Impact Measurement Interface** specification.
+
+### Example
+
+Concrete example of how we might model the carbon impact of a web server using an **Impact Graph**.
+
+```
+My Web Server Impact Graph
+```
+
+We first create an **Impact Graph** that describes the ontology we want to measure. This might be an application like this case which is a Web Server, but it also might be a campaign or even an individual user journey. It could even be used to measure the embodied carbon of physical devices. What we are proposing here isn't limited to just software.

```

Comment:
We first create an **Impact Graph** that describes the software boundary we want to measure. The software boundary is represented as an ontology.

Reply:
```suggestion
We first create an **Impact Graph** that describes the software boundary we want to measure. The software boundary is represented as an ontology. This might be an application like this case which is a Web Server, but it also might be a campaign or even an individual user journey. It could even be used to measure the embodied carbon of physical devices. What we are proposing here isn't limited to just software.
```